scoap3/modules/records/receivers.py

```diff
@@ -33,3 +35,9 @@ def index_after_commit(sender, changes):
                     # Record not found in ES
                     current_app.logger.warning('Record with id "%s" not found in ElasticSearch' %
                                                model_instance.json.get('control_number'))
+
+
+@after_record_delete.connect
+def delete_pid_after_record_deletion(sender, record):
+    pid = PersistentIdentifier.get_by_object('recid', 'rec', record.id)
+    pid.delete()
```

Comment:
I think you have to call `db.session.commit` using `from invenio_db import db`, to make the changes effective